tests/exchange/test_exchange.py

```diff
@@ -1076,7 +1110,13 @@ def test_create_order(default_conf, mocker, side, ordertype, rate, marketprice,
     exchange = get_patched_exchange(mocker, default_conf, api_mock, id=exchange_name)
 
     order = exchange.create_order(
-        pair='ETH/BTC', ordertype=ordertype, side=side, amount=1, rate=200)
+        pair='ETH/BTC',
+        ordertype=ordertype,
+        side=side,
+        amount=1,
+        rate=200,
+        leverage=3.0
```

Comment:
i must say, i find this pretty pointless.

While it adds leverage to the call - it's not checking anything in relation to that - so just with this, we won't know if set_leverage was called, if "leverage" was in the arguments.
Anyway i think we'll need to have both cases - one with leverage, and one without - as i would expect the 2 to behave slightly differently.

Reply:
I tried to add `assert api_mock.create_order.call_args[0][5] == 3.0` to the test but it failed. I don't really know what to put in the test to test this

Reply:
which is why i think this should be omited / left to 1 (and maybe a different test can test leverage).

The "assert" will not work this way (leverage is not passed to `api.create_order()` directly).
For binance / ftx, you'll need to check if `set_leverage` was called.
for kraken, the test needs to be if it's contained in parameters. I THINK that's already covered by other tests ...

Reply:
Tried to check if they're called, it's not working, not really sure what i need to change

Reply:
in this test - it'll be pretty difficult.
best revert it to 1 (or omit the leverage parameter) and design a dedicated test for it (most likely per exchange - if it doesn't exist already).

The way you call it in THIS test is actually wrong, anyway (potentially should fail - but it's a state users shouldn't be able to reach, so ...). 
`default_conf` assumes spot markets (i'd leave that as is) - but you're calling leverage (which would assume non-spot markets) (which is probably why it ain't called.

Reply:
I just added it to the same test, it works, but it means more tests are run that necessary. I can create another test for it if you want, but it will mean duplicate code, because the code will be almost exactly the same